app/src/components/Header/wallet/evm/Content.tsx

```diff
@@ -67,6 +69,13 @@ const ContentBase = (props: ContentProps) => {
     }
   };
 
+  const {
+    target: { isEVM },
+  } = useDeploymentTarget();
+
+  const shouldShowAddButton =
+    !(isEVM && connectType === 'WALLETCONNECT') && onAddToken;
```

Comment:
this logic would read better if it was 

isEVM && connectType === 'METAMASK' && onAddToken

Reply:
I guess we dont need to check for isEVM here because the actual component is the EVM specific component...

Perhaps we add a "EvmConnectTypeSwitch" component like we have with other switch style components so we can expand in the future like

<EvmConnectTypeSwitch metaMask={...} walletConnect={...} />

then this would become

{onAddToken && <EvmConnectTypeSwitch metaMask={...} />}

Reply:
✅ https://github.com/Anchor-Protocol/anchor-web-app/pull/314/commits/801b87386ed76567b496ffcf2834cb7d041bc0c3